Branch-SDK/Branch-SDK/BNCStrongMatchHelper.m

```diff
@@ -112,11 +112,23 @@ - (void)presentSafariVCWithBranchKey:(NSString *)branchKey {
         
         // Must be on next run loop to avoid a warning
         dispatch_async(dispatch_get_main_queue(), ^{
-            [windowRootController presentViewController:safController animated:YES completion:^{
-                [self.secondWindow.rootViewController dismissViewControllerAnimated:YES completion:NULL];
-                [BNCPreferenceHelper preferenceHelper].lastStrongMatchDate = [NSDate date];
-                self.requestInProgress = NO;
-            }];
+            // Add the safari view controller using view controller containment
+            [windowRootController addChildViewController:safController];
+            [windowRootController.view addSubview:safController.view];
+            [safController didMoveToParentViewController:windowRootController];
+            
+            // Remove the safari view controller from view controller containment
```

Comment:
Re delegate: I spent a while on that. You're welcome to attempt that, but I was unable to find a workable solution.

Re second window lower: if you figure that out, great. We haven't yet seen these issues, so it's hard for us to know what will fix them. I really appreciate you helping with this.